pkg/lib/config/secret_update_instruction.go

```diff
@@ -833,15 +833,21 @@ func (i *SMSProviderSecretsUpdateInstruction) set(currentConfig *SecretConfig) (
 			AuthToken:           i.SetData.TwilioCredentials.AuthToken,
 			MessagingServiceSID: i.SetData.TwilioCredentials.MessagingServiceSID,
 		}
-		upsert(TwilioCredentialsKey, twilioCredentials)
+		err := upsert(TwilioCredentialsKey, twilioCredentials)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	if i.SetData.CustomSMSProvider != nil {
 		customSMSProviderConfig := CustomSMSProviderConfig{
 			URL:     i.SetData.CustomSMSProvider.URL,
 			Timeout: i.SetData.CustomSMSProvider.Timeout,
 		}
-		upsert(CustomSMSProviderConfigKey, customSMSProviderConfig)
+		err := upsert(CustomSMSProviderConfigKey, customSMSProviderConfig)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I can think of two actions we need to support. The implementation currently implements `set` only.

The actions are

1. Remove all SMS provider secrets. This action is performed when the developer disable custom SMS provider.
2. Upsert the selected provider secrets, and remove the secrets of non-selected ones.

Therefore, the implemented actions should be

- `upsert_specified_and_delete_unspecified`
- `delete_all`

Reply:
After a second thought it could still cause problem if secrets remaining when turning off custom sms gateway.
Let me change the behavior of `set`, so that it also cleanup secrets. `delete_all` can be done by setting null to all secrets.